src/map/lua/lua_baseentity.cpp

```diff
@@ -1741,6 +1741,46 @@ inline int32 CLuaBaseEntity::clearTargID(lua_State* L)
     return 0;
 }
 
+/************************************************************************
+*  Function: getPathPoint()
+*  Purpose : To get the current path point the entity is on
+*  Example : if npc:getPathPoint() == 3 then do stuff end
+*  Notes   : See Novalmauge in Bostaunieux Oubliette
+*  Notes   : currently used for pathfind.lua npcBasicPath
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getPathPoint(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+
+    uint16 point = m_PBaseEntity->GetPathPoint();
+
+    lua_pushnumber(L, point);
+    return 1;
+}
+
+/************************************************************************
+*  Function: setPathPoint()
+*  Purpose : To set the current path point the entity is on
+*  Example : npc:setPathPoint(pathPoint +1)
+*  Notes   : See Novalmauge in Bostaunieux Oubliette
+*  Notes   : currently used for pathfind.lua npcBasicPath
+************************************************************************/
+
+inline int32 CLuaBaseEntity::setPathPoint(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);
```

Comment:
If change the break to be only on players, we might be able to use this pathing method for mobs too~!

Reply:
yeah sounds good to me

Reply:
This has been done now